src/event_field_transformations.cpp

```diff
@@ -17,6 +25,29 @@ static std::vector<cata_variant> species_of_monster( const cata_variant &v )
     return result;
 }
 
+static std::vector<cata_variant> terrain_flag( const cata_variant &v )
+{
+    const std::set<std::string> &flags = v.get<ter_id>()->get_flags();
+    std::vector<cata_variant> result;
+    result.reserve( flags.size() );
+    for( const std::string &s : flags ) {
+        result.push_back( cata_variant( s ) );
+    }
+    return result;
+}
+
+static std::vector<cata_variant> terrain_to_swimming( const cata_variant &v )
```

Comment:
I'd probably call this `is_swimming_terrain`.  Naming something with a `is_` name is a nice way to signal that it returns bool.